### DISCUSSION TITLE
Multiple Start Events

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#75] [General]
Hi @ralphhanna,

I would have your opinion on the below scenario:

I would like start a workflow based on a trigger programmatically. Let's say the `trigger` is of type `string` whose values can be `A`, `B` , `C`, `D` or `E`.

![image](https://user-images.githubusercontent.com/55505611/203358456-d0674ce9-3785-4a7b-8c14-773a18a09ce1.png)

I would like to start the above example workflow **only** if the value of the `trigger` is `A` , `B` or `C`. Likewise based on the `trigger` value, I would like to start the workflows associated with the trigger.

Is it possible to achieve this programmatically using your engine?

Any help would be appreciated. Thanks!

**Comment by ralphhanna** [comment]
Hi,
I hope that I understand your question correctly,

BPMN-Server supports multiple start events, so when you start a process 
```js

   server.engine.start(processname, mydata, startNodeId);

```
So you need to provide the code that respond to the trigger and select the appropriate start node.

Another approach is to conditional flow, by have one start event that branches into 3 different actions. this would be the simplest approach.

A third options is to issue a 3 different messages for each separate process.

BPMN-Server does not support conditional start event, if that is what you meant.

Thanks,

**Reply by MaheshkumarSundaram** [reply]
Great. Thanks for the briefing.
